File: Python/Recursion-1/end_x.py

# given a string, compute recursively a new string where all the lowercase 'x' chars have been moved to the end of the string
def end_x(str):
	if not len(str):
		return ''
	if str[0] != 'x':
		return str[0] + end_x(str[1:])
	else:
		return end_x(str[1:]) + 'x'

print(end_x('xxre'))
print(end_x('xxhixx'))
print(end_x('xhixhix'))

File: Java/Recursion-1/endX.java

public class recursion1 {
	
	// given a string, compute recursively a new string where all the lowercase 'x' chars have been moved to the end of the string
	public String endX(String str) {
		
		if (str.isEmpty()) {
			return "";
		}
		
		if (str.charAt(0) != 'x') {
			return str.charAt(0) + endX(str.substring(1));
		}
		else {
			return endX(str.substring(1)) + "x";
		}
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.endX("xxre"));
		System.out.println(o.endX("xxhixx"));
		System.out.println(o.endX("xhixhix"));
		
	}

}
